app/modules/elocaps/cruds_elocaps.py

```diff
@@ -0,0 +1,271 @@
+                .where(
+                    (models_elocaps.Player.user_id == user_id)
+                    & (models_elocaps.GamePlayer.game_id == game_id),
+                )
+                .options(
+                    selectinload(models_elocaps.GamePlayer.game).selectinload(
+                        models_elocaps.Game.game_players,
+                    ),
+                ),
+            )
+        )
+        .scalars()
+        .first()
+    )
+
+
+async def user_game_validation(
+    db: AsyncSession,
+    game_player: models_elocaps.GamePlayer,
+) -> None:
+    game_player.has_confirmed = True
+    await db.commit()
+
+
+async def end_game(db: AsyncSession, game_id: str) -> None:
+    game: models_elocaps.Game | None = (
+        (
+            await db.execute(
+                select(models_elocaps.Game)
+                .where(models_elocaps.Game.id == game_id)
+                .options(
+                    selectinload(models_elocaps.Game.game_players).selectinload(
+                        models_elocaps.GamePlayer.player,
+                    ),
+                ),
+            )
+        )
+        .scalars()
+        .first()
+    )
+    if game is None:
+        raise ValueError("This game does not exist")
+    for x in game.game_players:
+        x.player.elo += x.elo_gain
+    await db.commit()
+
+
+async def get_leaderboard(
+    db: AsyncSession,
+    game_mode: CapsMode,
+    count=10,
+) -> Sequence[models_elocaps.Player]:
+    result = await db.execute(
+        select(models_elocaps.Player)
+        .distinct()
+        .join(models_elocaps.GamePlayer)
+        .join(models_elocaps.Game)
+        .where(
+            not_(
+                select(models_elocaps.GamePlayer)
+                .correlate(models_elocaps.Game)
+                .where(
+                    (models_elocaps.GamePlayer.game_id == models_elocaps.Game.id)
+                    & not_(models_elocaps.GamePlayer.has_confirmed),
+                )
+                .exists(),
+            )
+            & (models_elocaps.Player.mode == game_mode),
+        )
+        .order_by(desc(models_elocaps.Player.elo))
+        .limit(count)
+        .options(selectinload(models_elocaps.Player.user)),
+    )
+    return result.scalars().all()
+
+
+async def get_winrate(
+    db: AsyncSession,
+    game_mode: CapsMode,
+    user_id: str,
+) -> float | None:
+    result = await db.execute(
+        select(models_elocaps.Game)
+        .join(models_elocaps.GamePlayer)
+        .join(models_elocaps.Player)
+        .where(
+            (models_elocaps.Game.mode == game_mode)
+            & (models_elocaps.Player.user_id == user_id),
+        )
+        .options(
+            selectinload(models_elocaps.Game.game_players).selectinload(
+                models_elocaps.GamePlayer.player,
+            ),
+        ),
+    )
+    games: list[models_elocaps.Game] = [
+        x for x in result.scalars().all() if x.is_confirmed
+    ]
```

Comment:
```suggestion
    # We need to filter confirmed games in a loop instead of an SQL condition
    # because `is_confirmed` is a computed property
    games: list[models_elocaps.Game] = [
        x for x in result.scalars().all() if x.is_confirmed
    ]
```